titiler/stacapi/dependencies.py

```diff
@@ -84,36 +86,73 @@ def OutputType(
     return accept_media_type(request.headers.get("accept", ""), accepted_media)
 
 
+class APIParams(TypedDict, total=False):
+    """STAC API Parameters."""
+
+    api_url: str
+    headers: Optional[Dict]
+
+
+def STACApiParams(
+    request: Request,
+) -> APIParams:
+    """Return STAC API Parameters."""
+    return APIParams(
+        api_url=request.app.state.stac_url,
+    )
```

Comment:
Here user could add `Auth` and also set `headers` to be forwarded to the stacAPI endpoints